### PR TITLE
SPCTR-3458 - Spec AI: Admin Migration to Tools

### PR DESCRIPTION
**Pull request by bsf-zanev**
### Description
Moved Spec AI to Tools Section, and created new Directory Structure for Spec AI.

### Types of changes
New feature (non-breaking change which adds functionality

### How has this been tested?
Extensively tested in the editor, in the tools page, as well as with the SCS.

### Checklist:
- [ ] My code is tested
- [ ] My code follows accessibility standards 
- [ ] My code has proper inline documentation 
- [ ] I've included any necessary tests 
- [ ] I've included developer documentation 
- [ ] I've added proper labels to this pull request

### CODE REVIEW

File: spec-ai/classes/class-spec-helpers.php [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+	 * @return array The Spec AI Response.
+	 */
+	public static function get_scs_response( $endpoint = 'chat/completions' ) {
+		// Get the Auth Token from the Spec AI Settings.
+		$spec_auth_token = self::get_decrypted_auth_token();
+
+		// If the Spec AI Auth Token is not set, then abandon ship.
+		if ( empty( $spec_auth_token ) || ! is_string( $spec_auth_token ) ) {
+			return array(
+				'error' => 'The Spec AI Auth Token is not set.',
```
Comment (adi3890) [inline spec-ai/classes/class-spec-helpers.php:127]: should be translatable missing multiple occurances.

File: spec-ai/classes/class-spec-helpers.php [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+			);
+		}
+
+		// Set the API URL.
+		$api_url = 'https://credits.startertemplates.com/api/' . $endpoint;
+
+		// Get the response from the endpoint.
+		$response = wp_remote_post(
```
Comment (adi3890) [inline spec-ai/classes/class-spec-helpers.php:135]: wp_remote_post can also return WP_error that need to be handled.

File: spec-ai/admin/core/class-admin-configurations.php [hunk 3]
```diff
@@ -0,0 +1,293 @@
...
+	 */
+	public function admin_settings_ajax() {
+		// Verify the nonce.
+		check_ajax_referer( 'spec_ai_admin_settings_nonce', 'nonce' );
+
+		$spec_options = Spec_Helpers::get_admin_settings_option( 'spec_ai_settings', array() );
+
+		// Check if the enable_spec_ai is set.
+		if ( ! empty( $_POST['enable_spec_ai'] ) && is_string( $_POST['enable_spec_ai'] ) ) {
+			$spec_options['enabled'] = ( 'yes' === $_POST['enable_spec_ai'] ) ? true : false;
```
Comment (adi3890) [inline spec-ai/admin/core/class-admin-configurations.php:109]: sanitization is missing

File: spec-ai/admin/dashboard-app/src/screens/routes/Welcome.js [hunk 4]
```diff
@@ -0,0 +1,298 @@
...
+			settingsData.append( 'enable_spec_ai', invertedSpecEnabled );
+			fetch( spec_ai_react.ajax_url, {
+				method: 'POST',
+				credentials: 'same-origin',
+				body: settingsData,
+			} )
+			.then( ( resp ) => resp.json() )
+			.then( ( data ) => {
+				if ( data.success ) {
+					setIsSpecEnabled( data.data?.enabled ? 'yes' : 'no' );
```
Comment (adi3890) [inline spec-ai/admin/dashboard-app/src/screens/routes/Welcome.js:46]: This may not be required.

File: spec-ai/class-spec-ai.php [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+	 * Setup the required classes.
+	 * 
+	 * @since x.x.x
+	 * @return void
+	 */
+	public function setup_classes() {
+
+		if ( is_admin() ) {
+			// Kickin' off the Admin Loader.
+			Admin_Loader::get_instance();
```
Comment (adi3890) [inline spec-ai/class-spec-ai.php:123]: this can be by-passed for now. will update it accordingly as per library requirement.

**Review by adi3890 — COMMENTED** [review]
Multiple things need to be fixed and code need to be moved to PRO.

File: admin-core/assets/src/dashboard-app/pages/settings/Settings.js [hunk 6]
```diff
@@ -29,7 +29,7 @@ import ContainerGlobalElementsGap from '@DashboardApp/pages/settings/ContainerGl
 import MyAccount from '@DashboardApp/pages/settings/MyAccount';
 import InstagramUsers from '@DashboardApp/pages/settings/block-settings/InstagramUsers';
 import InheritFromTheme from '@DashboardApp/pages/settings/InheritFromTheme';
-import SpectraAISettings from '@DashboardApp/pages/settings/spectra-ai-settings/SpectraAISettings';
+// import SpectraAISettings from '@DashboardApp/pages/settings/spectra-ai-settings/SpectraAISettings';
```
[the remark this answers is not in the document]
Comment (bsf-zanev) [reply admin-core/assets/src/dashboard-app/pages/settings/Settings.js:32]: I've kept commented code just for the brand voice.

We can remove it completely if desirable, and just keep the brand voice file unused.

File: classes/class-uagb-loader.php [hunk 7]
```diff
@@ -126,7 +126,12 @@ public function define_constants() {
 		 * @return void
 		 */
 		public function loader() {
+			// If the UAGB directory is not defined, then bail.
+			if ( ! defined( 'UAGB_DIR' ) ) {
```
[the remark this answers is not in the document]
Comment (bsf-zanev) [reply classes/class-uagb-loader.php:130]: This was added to avoid PHPStan issues when we had imported Spec AI, but it has been removed now since the admin DIR has beem moved.

File: admin-core/inc/admin-menu.php [hunk 8]
```diff
@@ -232,6 +232,16 @@ public function styles_scripts() {
 			'description' => get_bloginfo( 'description' ),
 		);
 
+		// Temporary Check for Spec Authorization.
+		$spec_ai_options    = \UAGB_Admin_Helper::get_admin_settings_option( 'spec_ai_settings', array() );
+		$is_spec_authorized = (
+			! empty( $spec_ai_options )
```
[the remark this answers is not in the document]
Comment (bsf-zanev) [reply admin-core/inc/admin-menu.php]: This code has been safely removed.

File: spec-ai/admin/core/class-admin-configurations.php [hunk 9]
```diff
@@ -0,0 +1,293 @@
...
+
+	/**
+	 * Setup the Admin Settings Ajax.
+	 *
+	 * @since x.x.x
+	 * @return void
+	 */
+	public function admin_settings_ajax() {
+		// Verify the nonce.
+		check_ajax_referer( 'spec_ai_admin_settings_nonce', 'nonce' );
```
[the remark this answers is not in the document]
Comment (bsf-zanev) [reply spec-ai/admin/core/class-admin-configurations.php]: This has been done, but is now in Pro.

File: spec-ai/admin/core/class-admin-configurations.php [hunk 10]
```diff
@@ -0,0 +1,293 @@
...
+	 */
+	public function render_dashboard() {
+
+		// Check if Spec is Authorized, then either render the settings page or the auth screen.
+		$spec_options   = Spec_Helpers::get_admin_settings_option( 'spec_ai_settings', array() );
+		$menu_page_slug = $this->menu_slug;
+
+		if ( ! Spec_Helpers::is_spec_authorized() ) {
+			// If Spec is not authorized, render the auth screen.
+			Admin_Views::render_admin_auth_markup( $menu_page_slug );
```
[the remark this answers is not in the document]
Comment (bsf-zanev) [reply spec-ai/admin/core/class-admin-configurations.php]: Reload will be required, since these two pages are not rendered directly one after another - the auth page had a wrapper that adds CSS to hide the unrequired core WP elements on purpose.

File: spec-ai/admin/dashboard-app/src/screens/DashboardRouter.js [hunk 11]
```diff
@@ -0,0 +1,23 @@
+import { __ } from '@wordpress/i18n';
+import { BrowserRouter as Router, Route, Switch } from 'react-router-dom';
+import AdminHeader from '@DashboardApp/common/components/AdminHeader';
+// import MainNav from '@DashboardApp/common/components/header/MainNav';
```
[the remark this answers is not in the document]
Comment (bsf-zanev) [reply spec-ai/admin/dashboard-app/src/screens/DashboardRouter.js]: I've removed the same, we'll add it later when the need arises.

File: spec-ai/classes/class-spec-helpers.php (hunk 1 above)
Comment (bsf-zanev) [reply spec-ai/classes/class-spec-helpers.php]: These string do not need to be translateable at the moment, but I've updated the same in pro in case we want to show the users the errors later on.

File: spec-ai/classes/class-spec-helpers.php (hunk 2 above)
Comment (bsf-zanev) [reply spec-ai/classes/class-spec-helpers.php]: That is handled in Pro now.

File: spec-ai/admin/core/class-admin-configurations.php (hunk 3 above)
Comment (bsf-zanev) [reply spec-ai/admin/core/class-admin-configurations.php]: This has been done in Pro.

File: spec-ai/admin/dashboard-app/src/screens/routes/Welcome.js (hunk 4 above)
Comment (bsf-zanev) [reply spec-ai/admin/dashboard-app/src/screens/routes/Welcome.js]: This has been improved, the toggle will only be reverted in the catch.

**Review by adi3890 — APPROVED** [review]
